lib/node_modules/@stdlib/assert/deep-has-own-property/lib/d_has_own.js

```diff
@@ -12,24 +12,23 @@ var hasOwnProp = require( '@stdlib/assert/has-own-property' );
 * Returns a boolean.
 *
 * @private
-* @param {ObjectLike} objLikeVal - input object
-* @param {Array} properties - list of properties defining a key path
-* @returns {*} boolean
+* @param {ObjectLike} obj - input object
+* @param {Array} props - list of properties defining a key path
+* @returns {boolean} boolean indicating whether an object has a nested property
 */
-function deepHasOwn( objLikeVal, properties ) {
-	var propLength = properties.length;
-	var value = objLikeVal;
+function deepHasOwn( obj, props ) {
+	var len = props.length;
 	var hasOwn;
-	var propInd;
-	for ( propInd = 0; propInd < propLength; propInd++ ) {
+	var i;
+	for ( i = 0; i < len; i++ ) {
 		if (
-			isObjectLike( value ) &&
-			hasOwnProp( value, properties[ propInd ] )
+			isObjectLike( obj ) &&
+			hasOwnProp( obj, props[ i ] )
 		) {
-			value = value[ properties[ propInd ] ];
+			obj = obj[ props[ i ] ];
 			hasOwn = true;
```

Comment:
This assignment is not necessary. As the loop will exit early only if we are unable to satisfy the condition. If we are able to iterate through the entire loop, the property exists. Hence, we can remove this assignment and, on L35, simply have `return true;`.